### PR TITLE
Only return a fix if the assembly identity matches the diagnostic mes…

### PR DESCRIPTION
**Pull request by jmarolf**
…sage

### CODE REVIEW

**Comment by jmarolf** [comment]
Tagging for Review: @Pilchie @jasonmalinowski @balajikris @rchande @brettfo @dpoeschl

File: src/Features/Core/AddMissingReference/AbstractAddMissingReferenceCodeFixProvider.cs [hunk 1]
```diff
@@ -95,18 +96,18 @@ private static IEnumerable<ITypeSymbol> GetTypesFromSymbol(ISymbol symbol)
...
         /// it is the assembly we want to add.
         /// </summary>
-        private static AssemblyIdentity GetAssemblyIdentity(IEnumerable<ITypeSymbol> types)
+        private static AssemblyIdentity GetAssemblyIdentity(IEnumerable<ITypeSymbol> types, string message)
         {
             foreach (var type in types)
             {
                 var identity = type?.GetBaseTypesAndThis().OfType<IErrorTypeSymbol>().FirstOrDefault()?.ContainingAssembly?.Identity;
-                if (identity != null)
+                if (identity != null && message.Contains(identity.ToString()))
```
Comment (jasonmalinowski) [inline src/Features/Core/AddMissingReference/AbstractAddMissingReferenceCodeFixProvider.cs]: Wasn't message parsing the entire thing we were trying to get out of in the first place?

**Comment by jasonmalinowski** [comment]
This is just screaming even more and more for an API or a better way to get this diagnostic information. What alternatives do we have?

**Comment by jmarolf** [comment]
This is a difficult design problem.

1.) Diagnostics could provide some property bag of info that we could retrieve.  The difficulty is how to make it serializeable.

2.) The compiler could expose [MissingMetadataTypeSymbol](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/MissingMetadataTypeSymbol.cs) which would at lease allow for better filtering logic of error messages.  Exposing this type would be problematic as we don't expose any of its parent types.

3.) We create an entirely new API that attempts to give us an [IErrorTypeSymbol](http://source.roslyn.io/#Microsoft.CodeAnalysis/Symbols/IErrorTypeSymbol.cs) for a given diagnostic. Unfortunately we will still have to check the assembly identity.

**Comment by jmarolf** [comment]
@jaredpar I don't think 1 can happen, would we be ok with something like 2 or 3?

**Pull request merged**